assemblies/static/src/main/resources/static/custom/js/sparkl.js

```diff
@@ -1,423 +1,454 @@
-
-
-    /* edit data of table  */
-    var editable = {
-        name: "editable",
-        label: "Editable",
-        defaults: {
-            action: function(v, st) {
-                Dashboards.log(v);
-            }
-        },
-        init: function() {
-
-            // Register this for datatables sort
-            var myself = this;
-            $.fn.dataTableExt.oSort[this.name + '-asc'] = function(a, b) {
-                return myself.sort(a, b);
-            };
-            $.fn.dataTableExt.oSort[this.name + '-desc'] = function(a, b) {
-                return myself.sort(b, a);
-            };
-        },
-        sort: function(a, b) {
-            return this.sumStrArray(a) - this.sumStrArray(b);
-        },
-
-        implementation: function(tgt, st, opt) {
-            var t = $(tgt);
-            var value = st.value;
-            var text = $("<input/>").attr({
-                'value': value,
-                'type': 'text',
-                'class': 'editBox'
-            })
-                .keyup(function(event) {
-                    if (event.keyCode == 13) {
-                        opt.action($(this).val(), st);
-                    }
-                    /*var idx = this.parentNode.parentNode.rowIndex;
-      metadataParam[idx-1][1] = $(this).val();*/
-                    var obj = this.parentNode.parentNode.children[0].textContent;
-                    metadataParam[obj.toString()] = $(this).val();
-                });
-
-            t.empty();
-            t.append(text);
-        }
-    };
-    Dashboards.registerAddIn("Table", "colType", new AddIn(editable));
+        $buttonContainer.append($button);
+      });
+
+      $(tgt).empty().append($buttonContainer);
+    }
+  };
+  Dashboards.registerAddIn("Table", "colType", new AddIn(actionButtonsOpts));
+
+
+  /* edit data of table  */
+  var editable = {
+    name: "editable",
+    label: "Editable",
+    defaults: {
+      action: function (v, st) {
+        Dashboards.log(v);
+      }
+    },
+
+    init: function () {
+      // Register this for datatables sort
+      var myself = this;
+
+      $.fn.dataTableExt.oSort[this.name + '-asc'] = function (a, b) {
+        return myself.sort(a, b);
+      };
+
+      $.fn.dataTableExt.oSort[this.name + '-desc'] = function (a, b) {
+        return myself.sort(b, a);
+      };
+    },
+
+    sort: function (a, b) {
+      return this.sumStrArray(a) - this.sumStrArray(b);
+    },
+
+    implementation: function (tgt, st, opt) {
+      var t = $(tgt);
+      var value = st.value;
+      var text = $("<input/>")
+        .attr({
+          'value': value,
+          'type': 'text',
+          'class': 'editBox'
+        })
+        .keyup(function (event) {
+          if (event.keyCode === 13) {
+            opt.action($(this).val(), st);
+          }
+
+          var obj = this.parentNode.parentNode.children[0].textContent;
+
+          var metadataParam = metadataParam || [];
```

Comment:
Are you sure? Shouldn't the variable be declared elsewhere?

Reply:
It doesn't appear to be (did some searchs in the code), but I will revert it just to play it safe